src/repository/Repository.ts

```diff
@@ -55,16 +55,41 @@ class Repository {
         const guildChannels = guild.channels.valueOf();
         const userList = await guild.members.fetch({user: userIDList});
         const userKarmaList = karmaInfo.map(info => {
-            const userName = userList.get(info.userID).user.username;
+            const user = userList.get(info.userID).user;
             return {
-                user: userName,
+                user,
                 karma: info.karma,
-                channel: guildChannels.get(info.channelID)?.name
+                channel: guildChannels.get(info.channelID)
             }
         });
         return userKarmaList.sort((a, b) => b.karma - a.karma);
     }
 
+    public async getUserKarmaInfo(userID: string): Promise<UserKarmaInfo[] | undefined> {
+        const keys = await this.getKeysAsync(`karmaInfo:${userID}:*`);
+        const karmaInfo: KarmaInfo[] = await Promise.all(keys.map(async (key) => {
+            const karma = await this.getAsync(key);
+            const [_, userID, channelID] = key.split(':');
+            return {
+                userID,
+                channelID,
+                karma: parseInt(karma)
```

Comment:
Do we have an ESLint setup? If not, we should add it, in a separate PR.

If we're using `parseInt`, it's always better to add second argument, i.e. `parseInt(karma, 10)`.

Or, we can use `Number(karma)`.

Reply:
Cool, will add ESlint next (y)